lib/internal/modules/helpers.js

```diff
@@ -298,14 +299,61 @@ function getBuiltinModule(id) {
   return normalizedId ? require(normalizedId) : undefined;
 }
 
+/** @type {import('internal/util/types')} */
+let _TYPES = null;
+/**
+ * Lazily loads and returns the internal/util/types module.
+ */
+function lazyTypes() {
+  return _TYPES ??= require('internal/util/types');
+}
+
+let DECODER = null;
+/**
+ * @param {string | ArrayBufferView | ArrayBuffer} body - The body to check.
+ * @param {boolean} allowString - Whether or not to allow a string as a valid buffer source.
+ * @param {string} hookName - The name of the hook being called.
+ * @throws {ERR_INVALID_RETURN_PROPERTY_VALUE} If the body is not a buffer source.
+ */
+function assertBufferSource(body, allowString, hookName) {
+  if (allowString && typeof body === 'string') {
+    return;
+  }
+  const { isArrayBufferView, isAnyArrayBuffer } = lazyTypes();
+  if (isArrayBufferView(body) || isAnyArrayBuffer(body)) {
+    return;
+  }
+  throw new ERR_INVALID_RETURN_PROPERTY_VALUE(
+    `${allowString ? 'string, ' : ''}array buffer, or typed array`,
+    hookName,
+    'source',
+    body,
+  );
+}
+
+/**
+ * Converts a buffer or buffer-like object to a string.
+ * @param {string | ArrayBuffer | ArrayBufferView} body - The buffer or buffer-like object to convert to a string.
+ * @returns {string} The resulting string.
+ */
+function stringify(body) {
+  if (typeof body === 'string') { return body; }
+  assertBufferSource(body, false, 'load');
+  const { TextDecoder } = require('internal/encoding');
+  DECODER ??= new TextDecoder();
+  return DECODER.decode(body);
+}
+
```

Comment:
Moving these functions is an unrelated change. It also might have unintended consequences regarding circular references, which plague the ESM loader; so this change needs to happen in its own PR. cc @joyeecheung

Reply:
I was actually able to revert this change